src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleArray.cs

```diff
@@ -54,35 +54,29 @@ public static partial class JsonSerializer
 
             state.Current.CollectionPropertyInitialized = true;
 
-            if (state.Current.JsonClassInfo.ClassType == ClassType.Value)
-            {
-                // Custom converter code path.
-                state.Current.JsonPropertyInfo.Read(JsonTokenType.StartObject, ref state, ref reader);
-            }
-            else
-            {
-                // Set or replace the existing enumerable value.
-                object value = ReadStackFrame.CreateEnumerableValue(ref reader, ref state);
+            // We should not be processing custom converters here.
+            Debug.Assert(state.Current.JsonClassInfo.ClassType != ClassType.Value);
```

Comment:
What does `ClassType.Value` have to do with custom converters?

Reply:
A custom converter is always `ClassType.Value` which means we need to do read-ahead to support. I believe the original line was a first attempt at handling this case but later replaced by the calling method.

Reply:
Can you add that as a comment. Seeing custom converters in the comment didn't make sense to me initially when we were talking about `ClassType`.